### PR TITLE
Add `SecretsProvider` implementations for AWS, GCP, Vault

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: catalog/secrets/cache/src/main/java/org/projectnessie/catalog/secrets/cache/CachingSecretsBackend.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                    return Math.max(0L, remaining);
+                  }
+
+                  @Override
+                  public long expireAfterUpdate(
+                      CacheKeyValue key,
+                      Map<String, String> value,
+                      long currentTimeNanos,
+                      @NonNegative long currentDurationNanos) {
+                    return expireAfterCreate(key, value, currentTimeNanos);
```
Comment (dimas-b) [inline catalog/secrets/cache/src/main/java/org/projectnessie/catalog/secrets/cache/CachingSecretsBackend.java:74]: nit: why not simply return `currentDurationNanos`? AFAIK, it is the value returned by `expireAfterCreate` at entry creation time.

**Review by dimas-b — APPROVED** [review]

File: catalog/secrets/aws/src/main/java/org/projectnessie/catalog/secrets/aws/AwsSecretsManager.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+
+public class AwsSecretsManager extends AbstractStringBasedSecretsManager {
+  private final SecretsManagerClient secretsManagerClient;
+  private final String prefix;
+
+  public AwsSecretsManager(
+      SecretsManagerClient secretsManagerClient, String prefix, Duration getSecretTimeout) {
```
Comment (dimas-b) [inline catalog/secrets/aws/src/main/java/org/projectnessie/catalog/secrets/aws/AwsSecretsManager.java:29]: `getSecretTimeout` is not used?

File: catalog/secrets/cache/src/main/java/org/projectnessie/catalog/secrets/cache/CachingSecretsBackend.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+    CacheKeyValue key = new CacheKeyValue(repositoryId, name, expires);
+
+    Secret fromCache =
+        cache.get(
+            key,
+            k -> {
+              @SuppressWarnings("unchecked")
+              Optional<Secret> loaded =
+                  (Optional<Secret>) backend.getSecret(name, secretType, secretJavaType);
```
Comment (dimas-b) [inline catalog/secrets/cache/src/main/java/org/projectnessie/catalog/secrets/cache/CachingSecretsBackend.java:125]: `backend.getSecret()` can block... Are we ok with that?

Comment (snazy) [reply catalog/secrets/cache/src/main/java/org/projectnessie/catalog/secrets/cache/CachingSecretsBackend.java:125]: yes

**Review by dimas-b — APPROVED** [review]

**Pull request merged**